h2o-core/src/main/java/hex/SubModelBuilder.java

```diff
@@ -0,0 +1,70 @@
+package hex;
+
+import org.apache.log4j.Logger;
+import water.H2O;
+import water.Job;
+import water.ParallelizationTask;
+import water.util.Log;
+
+/**
+ * Execute build of a collection of sub-models (CV models, main model) in parallel
+ * 
+ * This class is conceptually similar to CVModelBuilder and follows the same API. As opposed to CVModelBuilder
+ * it is not limited to building just CV models but can built a mixture of CV and main models.
+ * It also uses more efficient technique of model parallelization that works better when different sub-models
+ * (eg. CV folds) take vastly different time to complete.
+ * 
+ * It currently lacks prepare/finished feature of CVModelBuilder
```

Comment:
Is the ultimate goal to fully replace CVModelBuilder with this?
Looks like the prepare/finished feature mentioned here are just simple hooks implemented only for XGBoost currently.
I mean, would be great to merge those 2 classes in a future PR, and this one looks much cleaner.

Reply:
Yes, that is the goal to remove the existing CVModelBuilder - for that I will need more testing.